Makefile

```diff
@@ -1,22 +1,92 @@
+#! /usr/bin/make -f
+
+# Project variables.
+VERSION := development
+PROJECT_NAME := $(shell basename "$(PWD)")
 DATE := $(shell date '+%Y-%m-%dT%H:%M:%S')
-VERSION = development
 HEAD = $(shell git rev-parse HEAD)
 LD_FLAGS = -X github.com/tendermint/starport/starport/internal/version.Version='$(VERSION)' \
 	-X github.com/tendermint/starport/starport/internal/version.Head='$(HEAD)' \
 	-X github.com/tendermint/starport/starport/internal/version.Date='$(DATE)'
 BUILD_FLAGS = -mod=readonly -ldflags='$(LD_FLAGS)'
 
+# Go related variables.
+GOBASE := $(shell pwd)
+GOBIN := $(GOBASE)/dist
+GOCMD := $(GOBASE)/starport/cmd/starport
+GOCILINT := $(GOPATH)/bin/golangci-lint
+GORELEASER := $(GOPATH)/bin/goreleaser
+STARPORT := $(GOPATH)/bin/starport
+
+# For go 1.13-1.15 compatibility
```

Comment:
Starport requires at least 1.16 because we use the recently added `embed` feature.